### PR TITLE
Added support for non raw messages and aws credentials

### PR DESCRIPTION
**Pull request by runningman84**
based on https://github.com/runningman84/chef-monitor/commit/3f502c80fe7a0a08fc7cbd7af3007ac0d035ad18

### CODE REVIEW

**Comment by troyready** [comment]
@runningman84 I like it. Can you add some comments (maybe just a single one) to the message processing logic to explain what's happening (what condition equates to the message being raw or whatever)?

**Comment by autarchprinceps** [comment]
If the message is raw it contains the message_attributes array directly in the SQS message root. Therefore if this isn't the case it assumes this to be an encapsulated SNS message, in which case the message_attributes are in the message, which itself is a JSON (the original SNS message), but formated as a string.

File: lib/sensu/transport/snssqs.rb [hunk 1]
```diff
@@ -173,7 +184,19 @@ def receive_messages
...
+              tmp_body = JSON.parse msg.body
+              msg.body = tmp_body['Message']
+              msg.message_attributes = {}
+              tmp_body['MessageAttributes'].each do |name, value|
+                msg.message_attributes[name] = Aws::SQS::Types::MessageAttributeValue.new
+                msg.message_attributes[name].string_value = value['Value']
+                msg.message_attributes[name].data_type = 'String'
+              end
+            end
+            msg
```
Comment (autarchprinceps) [inline lib/sensu/transport/snssqs.rb:198]: A message containing the message_attributes directly is a raw message and can be handled the same way it is now.

File: lib/sensu/transport/snssqs.rb [hunk 2]
```diff
@@ -173,7 +184,19 @@ def receive_messages
             wait_time_seconds: @settings[:wait_time_seconds],
             max_number_of_messages: @settings[:max_number_of_messages],
           )
-          resp.messages
+          resp.messages.map do |msg|
+            if !msg.key? 'message_attributes'
```
Comment (autarchprinceps) [inline lib/sensu/transport/snssqs.rb:188]: A message without the message_attributes directly in the root is an encapsulated SNS message (JSON formated as string), it therefore needs two changes:
1. The message_attributes need to be extracted from the body
2. The body needs to be replaced with the original message.

**Review by autarchprinceps — APPROVED** [review]
As requested a little explanation.

**Comment by troyready** [comment]
@autarchprinceps thanks for the clarification.

@runningman84 can you put these as inline comments? I'm almost guaranteed to forget how the logic works here and would like to have it inline for future debugging (would be nice long-term to just have tests capturing the why of the logic but that's not on my radar to implement).

**Pull request merged**